Source/WebCore/rendering/RenderBox.h

```diff
@@ -610,7 +610,15 @@ override;
     bool hasVisualOverflow() const { return m_overflow && !borderBoxRect().contains(m_overflow->visualOverflowRect()); }
 
     virtual bool needsPreferredWidthsRecalculation() const;
-    virtual void computeIntrinsicRatioInformation(FloatSize& /* intrinsicSize */, double& /* intrinsicRatio */) const { }
+
+    struct AspectRatio {
+        double logicalWidth = 0;
```

Comment:
Should these be LayoutUnits at this point? Which means this struct is really just a LayoutSize?

Reply:
I believe when I first tried that there were certain tests that were starting to fail due to the ` m_value = clampToInteger(value * kFixedPointDenominator);` calculation but I will double check.

Reply:
which is completely fine, they just need rebaseline. We rely on LayoutUnit's precision (however limited it is) in layout.

Reply:
So changing to LayoutSize is mostly fine, the only test that is potentially concerning is `web-platform-tests/html/rendering/replaced-elements/attributes-for-embedded-content-and-images/img-aspect-ratio.html`

The test sets the width attribute to 0.8 and height attribute to 0.2 on an image element. The loss of information here eventually ends up in an aspect ratio of 4.25249169435216.

Storing the doubles helps avoid this issue, but the only concern I have with using them is that the LayoutUnit operators would also return doubles without checking for overflow (there is a WPT test checking this and using the LayoutSizes and the individual dimensions in this case results in a failure). Seems like it may need a little bit of work to get a correct solution

Reply:
After looking into this a bit more and playing around with the code/tests, I'm not sure if LayoutSize is the correct type to use in this context. I ended up using FloatSize, which seems to work fine, since the original values are crucial in the `img-aspect-ratio.html` test. If we convert these values into LayoutUnits we lose this information and I'm not sure we can pass the test without it, but let me know if any of you disagree or have other thoughts.